### PR TITLE
[SDP-877] Refactor SubmissionService interface and aggregate all Stellar Tx submission stuff in a single object called `SubmitterEngine`

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Move all objects related to Stellar transaction submission into a single object:
```go
type SubmitterEngine struct {
	HorizonClient horizonclient.ClientInterface
	LedgerNumberTracker
	SignatureService
	MaxBaseFee int
}
```

Also, refactor the SIgnatureService interface so it becomes more generic and safer:
1. It became safer by not receiving the full keypair for insertion but instead receiving the amount of accounts to be created. We don't want to pass secrets around, but instead rely on the sig service to create the secrets.
2. It became more generic by removing the currentLedgerNumber from the insert/delete method signatures. This was very implementation-specific.

### Why

So we can inject all the necessary services at once.

Also, to prepare the sig service for the VAULT implementation.

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [ ] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/dependencyinjection/main.go [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
 
-// dependenciesStoreMap var is the global map for all the service instances.
-var dependenciesStoreMap map[string]interface{} = map[string]interface{}{}
+import "sync"
+
+var (
+	// dependenciesStoreMap var is the global map for all the service instances.
+	dependenciesStoreMap = make(map[string]interface{})
+	// dependenciesStoreMutex is the mutex for the global map, used to make sure the map is thread safe.
+	dependenciesStoreMutex sync.Mutex
```
Comment (CaioTeixeira95) [inline internal/dependencyinjection/main.go:9]: Why don't we use the [`sync.Map`](https://pkg.go.dev/sync#Map) type here? Only a suggestion, no need to change it.

File: internal/transactionsubmission/engine/signature_service.go [hunk 2]
```diff
@@ -232,38 +233,54 @@ func (ds *DefaultSignatureService) SignFeeBumpStellarTransaction(ctx context.Con
...
+
+	currentLedgerNumber, err := ds.ledgerNumberTracker.GetLedgerNumber()
+	if err != nil {
+		return nil, fmt.Errorf("getting current ledger number: %w", err)
 	}
+	lockedToLedgerNumber := currentLedgerNumber + IncrementForMaxLedgerBounds
 
 	batchInsertPayload := []*store.ChannelAccount{}
-	for _, kp := range kps {
+	for range make([]interface{}, amount) {
```
Comment (CaioTeixeira95) [inline internal/transactionsubmission/engine/signature_service.go:248]: This syntax is not clear and may lead to misunderstanding.

```suggestion
	for i := 0; i < amount; i++ {
```

**Review by CaioTeixeira95 — APPROVED** [review]
🚀

File: internal/transactionsubmission/engine/signature_service.go (hunk 2 above)
Comment (marcelosalloum) [reply internal/transactionsubmission/engine/signature_service.go]: Done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/165/commits/d1289c36522c84aea1d84f898cc9b679bf5b2c47.

File: internal/dependencyinjection/main.go (hunk 1 above)
Comment (marcelosalloum) [reply internal/dependencyinjection/main.go]: Goof idea, done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/165/commits/d1289c36522c84aea1d84f898cc9b679bf5b2c47.

**Pull request merged**